gobblin-restli/gobblin-flow-config-service/gobblin-flow-config-service-server/src/main/java/org/apache/gobblin/service/FlowConfigResourceLocalHandler.java

```diff
@@ -171,6 +171,15 @@ public UpdateResponse updateFlowConfig(FlowId flowId, FlowConfig flowConfig, boo
       FlowConfig originalFlowConfig = getFlowConfig(flowId);
       originalFlowConfig.setSchedule(NEVER_RUN_CRON_SCHEDULE);
       flowConfig = originalFlowConfig;
+    } else if (isTriggerRequest(flowConfig)) {
```

Comment:
We do not need special handling to detect a "trigger" request, if we treat all Update FlowConfig operations as upserts i.e. merge the old FlowConfig with the config provided in the update request.

Reply:
Adding a few more thoughts:
For Collections resource type (as is the case with FlowConfigs), the recommended way to support updates to resources is via Partial Updates. At any rate, all full updates can be converted into partial updates.  We should deprecate the use of full updates of FlowConfigs and use partial updates as the only means for supporting updates.